sycl/include/sycl/ext/intel/experimental/online_compiler.hpp

```diff
@@ -42,7 +46,9 @@ class device_arch {
     gpu_cfl = gpu_gen9_5,
     gpu_gen11 = 4,
     gpu_icl = gpu_gen11,
-    gpu_gen12 = 5
+    gpu_gen12 = 5,
+    gpu_tgl = gpu_gen12,
+    gpu_tgllp = gpu_gen12
```

Comment:
I think this enum should eventually be deprecated and replaced with https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_intel_device_architecture.asciidoc#new-enumeration-of-architectures

Reply:
That enum is better, but IMO still not perfect solution because 
a) that enum may need special conversion to string, e.g. intel_gpu_icllp -> "icllp"; intel_gpu_acm_g12 -> "something-else'.
b) ocloc allows specifying not only 1 device, but a set of them using comma, e.g. "-device icllp,tgllp", and also it allows range of devices, e.g. "-device icllp:" which means something like icllp and newer.

Taking (a) and (b) into account, wouln't it be convenient to let user have that great flexibility provided by 'oclocl'? If Yes, the best ways is to let user specify the 'device_option' or 'device_filter' of std::string type.

Reply:
what problem are you trying to solve?

Reply:
Local goal in this PR: online_compiler became unusable after recent changes in GPU RT on Gen12 Windows - it stopped supporting "-device skl", cfl, etc. This PR changes the default from "skl" and adds mapping from enum gpu::gpu_gen12 to string "tgllp".

Bigger goal for future PRs should be (IMO): to allow users to compile for target device or group of devices they need and not be limited with some pre-defined enum (either the one used today, or this one https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_intel_device_architecture.asciidoc#new-enumeration-of-architectures)